Infinispan-operator-controller-manager crashes on 2.4.9

Hello, when patching infinispan operator from 2.4.8 to 2.4.9 the infinispan-operator-controller-manager logs this output and is stuck on crashloopbackoff

Log from the container:

```
panic: opensslcrypto: can't initialize OpenSSL : openssl: can't retrieve OpenSSL version
goroutine 1 [running]:
crypto/internal/backend.init.0()
/usr/lib/golang/src/crypto/internal/backend/openssl.go:65 +0x299
```

Same happen if we uninstall the operator completely and reinstalls it into fresh namespace.

We are running on openshift 4.14 and 4.16.

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,7 +12,7 @@ RUN go mod download
 
 # $rootfs will contain the libraries to manually export to the runtime container
 ENV rootfs=/tmp/rootfs
-RUN dnf install -y --installroot $rootfs openssl && \
+RUN dnf install -y --installroot $rootfs --setopt install_weak_deps=false --nodocs openssl && \
     dnf --installroot $rootfs clean all && \
     rm -rf $rootfs/var/cache/* $rootfs/var/log/dnf* $rootfs/var/log/yum.*
 
@@ -31,7 +31,8 @@ FROM registry.access.redhat.com/ubi9/ubi-micro
 COPY --from=build /bin/infinispan-operator /usr/local/bin/infinispan-operator
 # Open SSL library is not directly used by our operator, 
 # but it may be invoked by OCP using FIPS with GoLang 1.22
-COPY --from=build /tmp/rootfs/ /
+COPY --from=build /tmp/rootfs/usr/bin/openssl /usr/bin/openssl
+COPY --from=build /tmp/rootfs/lib64 /lib64
 # Define GOTRACEBACK to mark this container as using the Go language runtime
 # for `skaffold debug` (https://skaffold.dev/docs/workflows/debug/).
 ENV GOTRACEBACK=single

--- a/Dockerfile
+++ b/Dockerfile
@@ -10,6 +10,12 @@ COPY go.sum go.sum
 # and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
+# $rootfs will contain the libraries to manually export to the runtime container
+ENV rootfs=/tmp/rootfs
+RUN dnf install -y --installroot $rootfs openssl && \
+    dnf --installroot $rootfs clean all && \
+    rm -rf $rootfs/var/cache/* $rootfs/var/log/dnf* $rootfs/var/log/yum.*
+
 # Copy the go source
 COPY main.go main.go
 COPY api/ api/
@@ -23,6 +29,9 @@ RUN CGO_ENABLED=1 GOOS=linux GO111MODULE=on \
 
 FROM registry.access.redhat.com/ubi9/ubi-micro
 COPY --from=build /bin/infinispan-operator /usr/local/bin/infinispan-operator
+# Open SSL library is not directly used by our operator, 
+# but it may be invoked by OCP using FIPS with GoLang 1.22
+COPY --from=build /tmp/rootfs/ /
 # Define GOTRACEBACK to mark this container as using the Go language runtime
 # for `skaffold debug` (https://skaffold.dev/docs/workflows/debug/).
 ENV GOTRACEBACK=single

